### PR TITLE
Feature: new bundle return label rest api

### PR DESCRIPTION
**Pull request by DerCalli**
Implementierung RestApi bis zur Facade. Dort wird später über eine Bridge die Verbindung zu ReturnLaben (weiteres Bundle) hergestellt und das ReturnLabel über die DHL-API geholt.

### CODE REVIEW

File: bundles/return-labels-rest-api/src/FondOfOryx/Glue/ReturnLabelsRestApi/Proccesor/Reader/CompanyUnitAddressReader.php [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace FondOfOryx\Glue\ReturnLabelsRestApi\Proccesor\Reader;
+
+class CompanyUnitAddressReader implements CompanyUnitAddressReaderInterface
+{
+
```
Comment (julianzimmermann) [inline bundles/return-labels-rest-api/src/FondOfOryx/Glue/ReturnLabelsRestApi/Proccesor/Reader/CompanyUnitAddressReader.php:7]: u really need this empty class?

File: bundles/return-labels-rest-api/src/FondOfOryx/Glue/ReturnLabelsRestApi/Proccesor/ReturnLabelProcessor.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    /**
+     * @param ReturnLabelsRestApiTransfer $returnLabelsRestApiTransfer
+     *
+     * @return CompanyUnitAddressTransfer|null
+     */
+    protected function hasPermissionsToReadCompanyUnitAddress(
+        ReturnLabelsRestApiTransfer $returnLabelsRestApiTransfer
+    ): ?CompanyUnitAddressTransfer {
+        $companyUnitAddressTransfer = $this->client->findCompanyUnitAddressByUuid($returnLabelsRestApiTransfer);
```
Comment (julianzimmermann) [inline bundles/return-labels-rest-api/src/FondOfOryx/Glue/ReturnLabelsRestApi/Proccesor/ReturnLabelProcessor.php:87]: where is the permission check?

**Review by julianzimmermann — APPROVED** [review]

File: bundles/return-labels-rest-api/.editorconfig [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.json]
+indent_style = space
+indent_size = 2
+
+[*.yml]
+indent_style = space
+indent_size = 2
```
Comment (sourcecube) [inline bundles/return-labels-rest-api/.editorconfig:19]: Can be removed, this should be in monorepo root.

File: bundles/return-labels-rest-api/.gitignore [hunk 4]
```diff
@@ -0,0 +1,6 @@
+.idea
+
+.DS_Store
+
+vendor
+composer.lock
```
Comment (sourcecube) [inline bundles/return-labels-rest-api/.gitignore:6]: File is not necessary

File: bundles/return-labels-rest-api/LICENSE [hunk 5]
```diff
@@ -0,0 +1,9 @@
+The MIT License (MIT)
+
+Copyright (c) 2020 FOND OF GmbH
```
Comment (sourcecube) [inline bundles/return-labels-rest-api/LICENSE:3]: 2021

File: bundles/return-labels-rest-api/composer.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  "prefer-stable": true,
+  "autoload": {
+    "psr-4": {
+      "FondOfOryx\\": "src/FondOfOryx/"
+    }
+  },
+  "autoload-dev": {
+    "psr-4": {
+      "FondOfOryx\\": "tests/FondOfOryx/",
+      "Generated\\": "tests/Generated/"
```
Comment (sourcecube) [inline bundles/return-labels-rest-api/composer.json:28]: This needs to be `src/Generated`

File: bundles/return-labels-rest-api/composer.json [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    {
+      "name": "Pascal Fischer",
+      "email": "p-fischer@gmx.net"
+    }
+  ],
+  "require": {
+    "php": ">=7.4"
+  },
+  "require-dev": {
+    "fond-of-codeception/spryker": "dev-master",
```
Comment (sourcecube) [inline bundles/return-labels-rest-api/composer.json:15]: You can use version 1.0.0 I guess

File: bundles/return-labels-rest-api/src/FondOfOryx/Client/ReturnLabelsRestApi/ReturnLabelsRestApiDependencyProvider.php [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    /**
+     * @param Container $container
+     *
+     * @return Container
+     */
+    protected function addCompanyUserReferenceClient(Container $container): Container
+    {
+        $container->set(static::CLIENT_COMPANY_USER_REFERENCE_CLIENT, static function (Container $container) {
+            return new ReturnLabelsRestApiToCompanyUserReferenceClientBridge(
+                $container->getLocator()->companyUserReference()->getClient()
```
Comment (sourcecube) [inline bundles/return-labels-rest-api/src/FondOfOryx/Client/ReturnLabelsRestApi/ReturnLabelsRestApiDependencyProvider.php:53]: Is `getClient()` the right call here?

File: bundles/return-labels-rest-api/src/FondOfOryx/Client/ReturnLabelsRestApi/Zed/ReturnLabelsRestApiZedStub.php [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+     * @return CompanyUnitAddressTransfer|null
+     */
+    public function findCompanyUnitAddressByUuid(
+        ReturnLabelsRestApiTransfer $returnLabelsRestApiTransfer
+    ): ?CompanyUnitAddressTransfer {
+        return $this->zedRequestClient->call(
+            '/return-labels-rest-api/gateway/find-company-unit-address',
+            $returnLabelsRestApiTransfer
+        );
+    }
```
Comment (sourcecube) [inline bundles/return-labels-rest-api/src/FondOfOryx/Client/ReturnLabelsRestApi/Zed/ReturnLabelsRestApiZedStub.php:37]: Isn't there a client which exposes this query already? Could this be handled in the Zed Request which creates the label?

File: bundles/return-labels-rest-api/src/FondOfOryx/Shared/ReturnLabelsRestApi/ReturnLabelsConstants.php [hunk 10]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace FondOfOryx\Shared\ReturnLabel;
+
+interface ReturnLabelsConstants
+{
+
+}
```
Comment (sourcecube) [inline bundles/return-labels-rest-api/src/FondOfOryx/Shared/ReturnLabelsRestApi/ReturnLabelsConstants.php:8]: Can be removed if empty

File: dandelion.json [hunk 11]
```diff
@@ -1,97 +1,142 @@
...
-  },
-  "pathToTempDirectory": "/home/dandelion/project/tmp/",
-  "pathToSplitshLite": "/usr/local/bin/splitsh-lite"
+    "repositories": {
+        "customer-statistic": {
+            "url": "https://<GITHUB_TOKEN>@github.com/fond-of-oryx/customer-statistic.git",
+            "path": "bundles/customer-statistic/",
+            "version": "1.0.0"
+        },
+        "customer-statistics-rest-api": {
```
Comment (sourcecube) [inline dandelion.json:8]: Can you double check that your editor config is working properly and that JSON is indented with 2 spaces.

File: dandelion.json [hunk 12]
```diff
@@ -1,97 +1,142 @@
...
+        },
+        "mail-bcc": {
+            "url": "https://<GITHUB_TOKEN>@github.com/fond-of-oryx/mail-bcc.git",
+            "path": "bundles/mail-bcc/",
+            "version": "1.0.0"
+        },
+        "return-label-rest-api": {
+            "url": "https://<GITHUB_TOKEN>@github.com/fond-of-oryx/return-label-reast-api.git",
+            "path": "bundles/return-label-rest-api/",
+            "version": "1.0.0"
```
Comment (sourcecube) [inline dandelion.json:46]: This is old an incorrect, can you check that the branch is uptodate with `main`

File: codeception.yml [hunk 13]
```diff
@@ -19,6 +19,7 @@ include:
   - bundles/shipment-table-rate-data-import
   - bundles/splittable-checkout
   - bundles/tax-calculation-connector
+  - bundles/return-label-rest-api
```
Comment (sourcecube) [inline codeception.yml:22]: Typo ... in general this might be the better name

**Review by sourcecube — CHANGES_REQUESTED** [review]
I'm not sure about the plural `s` in the package name and there is a mix in naming in folder and package name

File: bundles/return-labels-rest-api/src/FondOfOryx/Client/ReturnLabelsRestApi/ReturnLabelsRestApiDependencyProvider.php (hunk 8 above)
Comment (DerCalli) [reply bundles/return-labels-rest-api/src/FondOfOryx/Client/ReturnLabelsRestApi/ReturnLabelsRestApiDependencyProvider.php]: Why not? Or should i implement it in my ZED layer (Facade-Bridge)?

**Pull request closed without merging**